pages/api/project.js

```diff
@@ -0,0 +1,35 @@
+// import { getServerSession } from "next-auth";
+import dbConnect from "../../db/connect";
+import Project from "../../db/models/Project";
+// import { authOptions } from "./auth/[...nextauth]";
+
+export default async function handler(request, response) {
+  await dbConnect();
+
+  // const session = await getServerSession(request, response, authOptions);
+  // const userId = session?.user.id;
+
+  // const postId = request?.body.id;
+
+  if (request.method === "GET") {
+    const projects = await Project.find().sort({ createdAt: -1 });
+    response.status(201).json(projects);
+  }
+
+  if (request.method === "POST") {
+    const { newProject } = request.body;
+    await Project.create(newProject);
+    response.status(200).json({ status: "project created" });
+  }
+  if (request.method === "PUT") {
+    const { id } = request.query;
+    await Project.findByIdAndUpdate(id, { $set: request.body });
+    response.status(200).json({ status: "project successfully updated" });
```

Comment:
afaik `status(201)` is for `POST` and `PUT` (but `status(200)` works for `PUT` too) and `status(200)` is for `GET`